compiler/rustc_trait_selection/src/solve/mod.rs

```diff
@@ -169,6 +164,8 @@ impl<'tcx> InferCtxtEvalExt<'tcx> for InferCtxt<'tcx> {
         let result = EvalCtxt {
             search_graph: &mut search_graph,
             infcx: self,
+            // Only relevant when canonicalizing the response.
+            max_input_universe: ty::UniverseIndex::ROOT,
```

Comment:
Wht's different between this and `InferCtxt::universe`?

Reply:
added a comment to this field. It's the same until we enter a binder inside of the query. either because the predicate is higher ranked, or when we relate binders.